app/views.py

```diff
@@ -84,6 +94,69 @@ def get_methods():
 def get_method(method):
     return jsonize(models.get_method(obfuscated_id=as_id(method)))
 
+
+@app.route('/prepare-multipart-upload', methods=['POST'])
+def prepare_file_upload():
+    pass
+
+
+@app.route('/upload-part', methods=['POST'])
+def upload_file():
+    part = request.files['file']
+
+    # part_number = request.form['part-number']
+    part_number = request.form['resumableChunkNumber']
+    # total_number_parts = request.form['total-parts']
+    total_number_parts = request.form['resumableTotalChunks']
+
+    # file_identifier = request.form['upload-id']
+    file_identifier = request.form['resumableIdentifier']
+    part_upload_dir = upload_dir(file_identifier)
```

Comment:
Is a full path to the scratch directory prepended to the file identifier earlier in the process? I'd hunt-and-peck, but hard to search on my phone. Sorry.